docs/source/reference-core.rst

```diff
@@ -656,6 +656,19 @@ Since all tasks are descendents of the initial task, one consequence
 of this is that :func:`run` can't finish until all tasks have
 finished.
 
+.. warning::
+
+  A return statement will not cancel the nursery if it still has tasks running.
+
+  .. code-block:: python
+  
+      with trio.open_nursery() as nursery:
+          nursery.start_soon(trio.sleep(5))
+
+          await trio.sleep(1)
+          return
+
+  This code will wait 5 seconds, NOT 1.
```

Comment:
How about:

* Make it a `.. note::`

* It's `start_soon(trio.sleep, 5)`, not `start_soon(trio.sleep(5))`

* I think we can simplify the example to just:

  ```python
  with trio.open_nursery() as nursery:
      nursery.start_soon(trio.sleep, 5)
      return
  ```

  This will wait 5 seconds for the child task to finish, and *then* the enclosing function will `return`.